github/git_commits.go

```diff
@@ -133,3 +149,49 @@ func (s *GitService) CreateCommit(ctx context.Context, owner string, repo string
 
 	return c, resp, nil
 }
+
+func createSignature(signingKey *openpgp.Entity, commit *createCommit) (string, error) {
+	if signingKey == nil || commit == nil {
+		return "", errors.New("createSignature: invalid parameters")
+	}
+
+	message, err := createSignatureMessage(commit)
+	if err != nil {
+		return "", fmt.Errorf("createSignature: createSignatureMessage error: %s", err.Error())
+	}
+
+	writer := new(bytes.Buffer)
+	reader := bytes.NewReader([]byte(message))
+	if err := openpgp.ArmoredDetachSign(writer, signingKey, reader, nil); err != nil {
+		return "", err
+	}
+
+	return writer.String(), nil
+}
+
+func createSignatureMessage(commit *createCommit) (string, error) {
+	if commit == nil || commit.Message == nil || commit.Author == nil {
+		return "", errors.New("createSignatureMessage: invalid parameters")
+	}
+
+	message := ""
```

Comment:
I'm just going to throw this out for your consideration, and I'm fine either way.
When I'm building up potentially long strings that are a collection of lines, I often like to use a slice of strings and then join them at the end.  I think it is much easier this way to write and also to maintain... because eventually someone will come along and add another line but forget to add their `\n`... so it looks something like this:

```go
var lines []string
...
lines = append(lines, fmt.Sprintf("...", ...))  // no \n
...
return strings.Join(lines, "\n"), err
```

Totally your call.  I prefer it, but this way is fine too.

Reply:
I like this, does make it easier to modify!